swap/src/bin/asb.rs

```diff
@@ -122,11 +122,11 @@ pub async fn main() -> Result<()> {
 
             // Initialize Monero wallet
             let monero_wallet = init_monero_wallet(&config, env_config).await?;
-            let monero_address = monero_wallet.get_main_address();
+            let monero_address = monero_wallet.lock().await.get_main_address();
```

Comment:
Note: Previously we had Mutex free access to the main address. We did not have to acquire a lock before because we are caching the main address in the struct. Not sure if this might cause some issues

Reply:
but I dont see another way

Reply:
Instead of an Mutex, we could use a RwLock to allow our code to call some of the functions without having to acquire a full Mutex? 

But not sure if needed.

Reply:
If acquiring the mutex for simple field access is a problem, we should split things like the main address into another struct and only wrap the wallet functions that use the client in a mutex.

Reply:
> If acquiring the mutex for simple field access is a problem, we should split things like the main address into another struct and only wrap the wallet functions that use the client in a mutex.

but thats what an RwLock is for: https://doc.rust-lang.org/std/sync/struct.RwLock.html

Reply:
can't you let the protected methods take a `&self mut`? This way we need a mut reference to call them?

If not: The current approach is also fine with me

Reply:
My worries are that, since we don't _need_ to use `&mut self`, it's likely that we'll mess up at some point and make some method that should be `&mut self` to `&self`. The rust compiler doesn't force us to use the correct reference type because all the methods on `wallet::Client` take `&self`.

Reply:
You convinced me 👍